### PR TITLE
Asset config [gql]

### PR DESCRIPTION
**Pull request by smackesey**
### Summary & Motivation

- Asset config resolvers for GQL
- Some typing improvements
- Modify bollinger example to use asset config

### How I Tested These Changes

By building the associated Dagit UI PR on top of this.

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #8163** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/8163" target="_blank">Graphite</a>  👈
    * **PR #8154** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/8154" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/8163?utm_source=stack-comment).

File: python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py [hunk 1]
```diff
@@ -226,6 +228,14 @@ def resolve_assetMaterializations(
             )
         ]
 
+    # TODO: Prob want a more efficient way of resolving this
+    def resolve_configField(self, _graphene_info) -> Optional[GrapheneConfigTypeField]:
+        op = self.resolve_op(_graphene_info)
```
Comment (clairelin135) [inline python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py:233]: I think we should try to pull the functionality of `resolve_op` into a helper function and then call the helper function to get back the op definition. In general, field resolvers of the same level should be independent of each other (so should not call one another).

File: python_modules/dagster-graphql/dagster_graphql/schema/dagster_types.py [hunk 2]
```diff
@@ -1,33 +1,48 @@
...
-from .config_types import GrapheneConfigType, to_config_type
+from .config_types import GrapheneConfigType, GrapheneConfigTypeUnion, to_config_type
 from .errors import (
     GrapheneDagsterTypeNotFoundError,
     GraphenePipelineNotFoundError,
     GraphenePythonError,
 )
 from .util import non_null_list
 
+GrapheneDagsterTypeUnion = Union[
```
Comment (clairelin135) [inline python_modules/dagster-graphql/dagster_graphql/schema/dagster_types.py:21]: Is this being used anywhere?

**Review by clairelin135 — APPROVED** [review]
Left a couple small comments but otherwise looking good to me! Thanks for all of the typechecks and annotations, makes the code much more readable!

File: python_modules/dagster-graphql/dagster_graphql/schema/dagster_types.py (hunk 2 above)
Comment (smackesey) [reply python_modules/dagster-graphql/dagster_graphql/schema/dagster_types.py:21]: Looks like not, good catch. I did this a while ago so probably part of some unfinished annotation work, but I've removed it for now.

File: python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py (hunk 1 above)
Comment (smackesey) [reply python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py]: Good call, I've now done this.

**Pull request merged**